\documentclass[11pt]{extarticle} %extarticle for fontsizes other than 10, 11 And 12
%\documentclass[11p]{article}

%%%%%%%%%%%%%%%%%%%%%%%% Packages %%%%%%%%%%%%%%%%%%%%%%%%
\usepackage{amscd}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{amsthm}
\usepackage{amsxtra}
\usepackage{bbold}
%\usepackage{bigints}
\usepackage{color}
\usepackage{dsfont}
\usepackage{enumerate}
\usepackage[mathscr]{eucal}
%\usepackage{fancyhdr}
\usepackage{float}
%\usepackage{fullpage} %% Dont use this for beamer presentations
\usepackage{geometry}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{indentfirst}
\usepackage{latexsym}
\usepackage{listings}
\usepackage{lscape}
\usepackage{mathtools}
\usepackage{microtype}
\usepackage{natbib}
\usepackage{pdfpages}
\usepackage{verbatim}
\usepackage{wrapfig}
\usepackage{xargs}
\DeclareGraphicsExtensions{.pdf,.png,.jpg, .jpeg, .bmp}

%%%%%%%%%%%%%%%%%%%%%%%% Commands %%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\Sup}{\textsuperscript}
\newcommand{\Exp}{\mathds{E}}
\newcommand{\Prob}{\mathds{P}}
\newcommand{\Z}{\mathds{Z}}
\newcommand{\Ind}{\mathds{1}}
\newcommand{\A}{\mathcal{A}}
\newcommand{\F}{\mathcal{F}}
\newcommand{\G}{\mathcal{G}}
\newcommand{\I}{\mathcal{I}}
\newcommand{\be}{\begin{equation}}
\newcommand{\ee}{\end{equation}}
\newcommand{\bes}{\begin{equation*}}
\newcommand{\ees}{\end{equation*}}
\newcommand{\union}{\bigcup}
\newcommand{\intersect}{\bigcap}
\newcommand{\Ybar}{\overline{Y}}
\newcommand{\ybar}{\bar{y}}
\newcommand{\Xbar}{\overline{X}}
\newcommand{\xbar}{\bar{x}}
\newcommand{\betahat}{\hat{\beta}}
\newcommand{\Yhat}{\widehat{Y}}
\newcommand{\yhat}{\hat{y}}
\newcommand{\Xhat}{\widehat{X}}
\newcommand{\xhat}{\hat{x}}
\newcommand{\E}[1]{\operatorname{E}\left[ #1 \right]}
%\newcommand{\Var}[1]{\operatorname{Var}\left( #1 \right)}
\newcommand{\Var}{\operatorname{Var}}
\newcommand{\Cov}[2]{\operatorname{Cov}\left( #1,#2 \right)}
\newcommand{\N}[2][1=\mu, 2=\sigma^2]{\operatorname{N}\left( #1,#2 \right)}
\newcommand{\bp}[1]{\left( #1 \right)}
\newcommand{\bsb}[1]{\left[ #1 \right]}
\newcommand{\bcb}[1]{\left\{ #1 \right\}}
\newcommand*{\permcomb}[4][0mu]{{{}^{#3}\mkern#1#2_{#4}}}
\newcommand*{\perm}[1][-3mu]{\permcomb[#1]{P}}
\newcommand*{\comb}[1][-1mu]{\permcomb[#1]{C}}

%%%%%%%%%%%%%%%%%%% To change the margins and stuff %%%%%%%%%%%%%%%%%%%
\geometry{left=1in, right=1in, top=1in, bottom=0.8in}
%\setlength{\voffset}{0.5in}
%\setlength{\hoffset}{-0.4in}
%\setlength{\textwidth}{7.6in}
%\setlength{\textheight}{10in}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}
%\SweaveOpts{concordance=TRUE}
\bibliographystyle{plain}  %Choose a bibliograhpic style

\title{Nanocoding and Fluoroscanning: Next Generation Precision Genomics}
\author{Subhrangshu Nandi\\
  Statistics PhD Student, \\
  Research Assistant,
  Laboratory of Molecular and Computational Genomics, \\
  University of Wisconsin - Madison}
\date{May 31, 2015}
%\date{}

\maketitle

%% \begin{center}
%% {\Large{Comparing different clustering techniques in analyzing gene-expression data}}\\
%% Subhrangshu Nandi\\
%% Stat 760: Multivariate Analysis\\
%% Project Proposal\\
%% November 25, 2014
%% \end{center}

\section*{Long term (big picture) goal}
To leverage ``Nanocoding'' and ``Fluoroscanning'' to quickly and economically analyze polymorphisms and mutations in human genomes. 

\section*{Fluoroscanning}
A critical step in producing nanocoded maps (Nmaps) is adding the dye Oxazole Yellow (YOYO) to the DNA molecules and then capturing the images of the molecules stretched on the positively charged surfaces, once they fluoresce. The dye molecules intercalates between the bases of the DNA molecules. Fluorescence properties of YOYO depends strongly on DNA sequence. In fact, quantum yield of YOYO bound to “GC” rich sequences is almost twice as large as those bound to “AT” rich sequences; binding constants also vary with sequence composition. Hence, parts of Nmaps that have been aligned to the same location on the genome should exhibit similar fluorescence intensity profiles. In fact, this is one of the first questions of our research. 

\subsection*{Aim 1: How reproducible are the fluorescence intensity profiles?}
To comprehend this aim, please refer to fig \ref{fig:Fig1}. Each panel is a curvilinear representation of the intensity of the DNA molecule images. This image is of Nmaps of {\emph{M.florum}} DNA molecules. The peaks of the curves correspond to high fluorescence intensity and vice versa. One point on a curve corresponds to one pixel on the image which in turn corresponds to approximately 200 base pairs (bp) of the DNA molecule. As is evident from the figure, this fragment is approximately 33.4 kb long, which produces images of approximately 158 pixels. The first aim is to quantify and establish this reproducibility (or consistency) of fluorescence intensity profiles of Nmaps aligned to the same location on the genomes. 

Before the microscopic photography these molecules are stretched on glass surfaces. Upon stretching more dye molecules might intercalate between the bases. Conversely, when more dye molecules intercalate between the bases, the DNA molecules could get more stretched. Hence, the observed intensity profiles are not free from errors. It is important to observe, quantify and statistically control for these scenarios.

%% \begin{figure}[H]
%%     \centering
%%     \begin{minipage}{.5\textwidth}
%%         \centering
%%         %\includegraphics[width=1.1\linewidth, height=0.4\textheight]{MF_chr1_frag24_Pixels156-1}
%%         \includegraphics[scale=0.5]{MF_chr1_frag24_Pixels156-1}
%%         %\caption{$dt=0.1$}
%%         \label{fig:prob1_6_2}
%%     \end{minipage}%
%%     \begin{minipage}{0.5\textwidth}
%%         \centering
%%         \includegraphics[scale=0.5]{MF_chr1_frag24_Pixels156-5}
%%         %\caption{$dt =$}
%%         \label{fig:prob1_6_1}
%%     \end{minipage}
%% \end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[scale=0.75]{MF_chr1_frag24_Pixels156-1}
	\caption{Intervals from 5 Nmaps that align to reference fragment 24 of {\emph{M.florum}}}
	\label{fig:Fig1}
\end{figure}

\subsection*{Aim 2: How unique are the fluorescence intensity profiles?}
If Aim 1 can be established, then the next question is whether regions of distinctive genomic sequence compositions have unique signature intensity profiles. If yes, then the intensity profile produced by a DNA molecule with structural variation(s) would not align with the expected intensity profile. This would complement the next generation sequencing techniques in detecting large scale genomic variations or damages in DNA. 

\subsection*{Curve registration and intensity profiles}
To address Aim 2, we need to estimate a consensus fluorescence intensity profile of any region on the genome from the intensity profiles  of all Nmaps aligned to that location. However, as can be observed in fig \ref{fig:Fig1}, the features of the curves (intensity profiles), the peaks and troughs are not aligned. This could be due to reasons yet to be fully understood. However, averaging these curves with misaligned features will lead to inaccurate estimate of the consensus fluorescence intensity profile. We are using a statistical technique called ``curve registration'' to align these features before estimating the consensus fluorescence intensity profile. 

\begin{figure}[H]
	\centering
	\includegraphics[scale=0.4]{MF_Frag30_46-85_Registered}
	\caption{Comparing sequence composition and intensity profiles}
	\label{fig:Fig2}
\end{figure}
Fig \ref{fig:Fig2} illustrates a region on the {\emph{M.florum}} genome, with the top panel representing genomic content and the bottom panel representing the consensus fluorescence intensity profile after registering the intensities from all the Nmaps aligned to that location on the genome. In the top panel, each vertical bar represents a 200 base pair window, with the colors representing the proportions of ``G'', ``C'', ``A'' and ``T'' in that window. 

\subsection*{Application: Detect heterozygosity}
One of the applications of Aim 2 is to identify large scale structural variations in genomes. One other application is detect heterozygosity. 
\begin{figure}[H]
    \centering
    \begin{minipage}{.5\textwidth}
        \centering
        %\includegraphics[width=1.1\linewidth, height=0.4\textheight]{MF_chr1_frag24_Pixels156-1}
        \includegraphics[scale=0.45]{MM52_Ch3_Registered_Class1.pdf}
        %\caption{$dt=0.1$}
        \label{fig:Fig3_1}
    \end{minipage}%
    \begin{minipage}{0.5\textwidth}
        \centering
        \includegraphics[scale=0.45]{MM52_Ch3_Registered_Class2.pdf}
        %\caption{$dt =$}
        \label{fig:Fig3_2}
    \end{minipage}
\caption{Detecting Heterozygosity in chr 3 of human genome, suffering from multiple myeloma}
\label{fig:Fig3}
\end{figure}

Fig \ref{fig:Fig3} is an example of using fluoroscanning to detect heterozygosity. Approximately half of the Nmaps aligned to this location on chr 3 produce the consensus fluorescence intensity profile on the left and the other half produce the one on the right. Since we don't yet have an expected consensus fluorescence intensity profile of this region, we do not know which set of Nmaps are deviant from the normal. However, this is a first step in the direction of addressing the question of detecting heterozygosity. 

\section*{Work in progress}
\begin{enumerate}
\item Develop a statistical framework for using curve registration to estimate the consensus fluorescence intensity profiles.
\item Use the above framework to estimate consensus fluorescence intensity profiles of regions with sufficient sample size. Establish a confidence levels of these consensus profiles. 
\item Develop a framework to classify Nmaps to regions on the genome whose consensus fluorescence intensity profiles have been estimated with high confidence levels.
\item Test the methodology on the human multiple myeloma sample data and detect structural variation.
\end{enumerate}
%\newpage
%\bibliography{bibTex_Reference}
%\bibliography{research}

\end{document}
