\documentclass[11pt]{extarticle} %extarticle for fontsizes other than 10, 11 And 12
%\documentclass[11p]{article}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Input header file 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\input{HeaderfileTexDocs}

\begin{document}
%\SweaveOpts{concordance=TRUE}
\bibliographystyle{plain}  %Choose a bibliograhpic style

\title{Fluoroscanning: Next generation precision genomics}
\author{Subhrangshu Nandi \\
%  Statistics PhD Student, \\
%  Research Assistant,
%  Laboratory of Molecular and Computational Genomics, \\
%  University of Wisconsin - Madison}
%\date{February 16, 2015}
\date{}
}

\maketitle


\section*{Abstract}
We will do this last and need to contour it for the journal we wish to publish with

%% The Human Genome Project (HGP), completed in 2003, is considered one of the greatest accomplishments of exploration in history of science. Since then thousands of genomes have been sequenced. However, no individual human genome has been annotated to completion. Nanocoding \cite{Jo_etal_2007_PNAS} (PNAS, 2007), developed by Laboratory of Molecular and Computational Genomics (LMCG), UW Madison, is a novel system for physically mapping genomes. It uses measurements of single DNA molecules to construct a high-resolution genome-wide restriction map, whose representation of genome structure complements genome sequences to yield biological insight. Staining the DNA molecules with cyanine dyes and imaging them are essential procedural requirements of nanocoding. The quantum yield of the fluorescence intensity of these stained molecules are DNA sequence dependent. In fact, for dye complexed with GC-rich DNA sequences the quantum yield are about twice as large as for those complexed with AT-rich sequences. Hence, regions with distinct genomic sequence compositions should exhibit unique fluorescence intensity profiles. Establishing the fluorescence intensity profiles of a genome would provide invaluable insights into its sequence compositions without having to sequence it. This is akin to producing a mp3 version of a wave file, an approximation of the true sequence, yet preserving the salient characteristics. This newly discovered technique is named ``Fluoroscanning''. Fluoroscanning can be applied to analyzing whole genomes. For example, imaged DNA molecules from the same region on a genome should exhibit similar intensity profiles, unless there has been a modification in the underlying genomic sequence. Fluoroscanning can also be used to identify heterozygotes and detect large scale structural variations as a result of cancer or other diseases. In the quest of establishing fluoroscanning as the next generation precision genomics we need to address some interesting statistical and computational challenges. In this thesis we apply some aspects of functional data analysis, like curve registration, to establish reproducibility and uniqueness of fluorescence intensity profiles. We also show how fluroscanning can detect structural variations and heterozygotes in a dataset collected from a cancer genome.

\newpage

\section*{Introduction}
\begin{itemize}
\item Background on high-throughput analysis of human genomes: SNPs, sequencing, the need to discover and catalogue structural variation in normal and cancer genomes.
\item The challenges facing large-scale genome analysis of human populations that are parsimonious, yet comprehensive. (PRECISION MEDICINE) These bottlenecks include: sample preparation, measurement, preprocessing of measurements into data sets, computational pipelines which run slowly ({\emph{we’ll need to address this critical issue here}}), and genomic resolution (power to discover new polymorphisms and mutations, yet enable tabulation of known genomic variants, across a broad spectrum of types - SNPs, SVs, CNVs.
\item Large, genomic molecules are ideal substrates, analyzed at the level of the individual molecule as ensembles. Avoidance of clones, or amplicons, which add errors and unneeded steps.
\item Background on OM, nanocoding, maybe nanaopore sequencing
\item Mapping systems are valuable, but must advance to encompass additional information that approaches sequencing. Efforts in this vein include single molecule analysis of melting curves, issues with such approaches center on necessary baroque implementations, precluding high-throughput and parsimony.
\item Fluorochrome binding and luminosity measurements are governed by sequence, which are leveraged by single molecule measurement approaches. 
\item A short history of dye/DNA interactions, going back to the 60’s (2 sentences).
\end{itemize}

\begin{enumerate}
\item Accordingly, we reasoned that given the spatial resolution of high N.A. fluorescence microscopy that usable ``signals'' are extractable from such data, despite scaling issues involving fluoro-binding considerations covering $~5$ bp and optical resolution spanning hundred of bps AND signal convolution considerations, involving the optical train.
\item It follows, that stretched, genomic DNA molecules are ideal analytes for Fluoroscanning. Use of Nanocoding provide a way to boot into Fluroscanning analysis and provide an ideal environment for cross-checking.
\end{enumerate}

REST OF INTRODUCTION WILL BE CRAFTED ONCE WE’VE WRITTEN OUR RESULTS SECTION. Here, we introduce and summarize our findings.

\section*{Results}
\begin{enumerate}
\item Stretched DNA molecules, presented on charged surfaces have varied fluorescence, as measured along molecular ``backbones''. Such initial findings prompted us to investigate further - are these signals reproducible? What errors are associated with such measurements?
\item Errors (physical/chemical): Stretching, dye concentration, surface conditions, photo bleaching effects of labeled punctates
\item Errors (machine vision): Overlapping molecules, backbone tracing, spurious fluorescence signals associated with molecules, surface/molecule interactions creating outlier signals
\item Model system ({\emph{M. florum}}) and data preprocessing. Small genome, allows facile acquisition of large data sets. Nanocoded datasets and alignment allow unambiguous identification of random, genomic DNA molecules.
\begin{itemize}
\item Describe M. florum dataset.
\item Normalization of Nmaps (Fscans). Relative signal used—method developed for normalizing signals, obviating error-prone “absolute” fluorescence intensity measurements.
\item Signals are noisy. They suffer from errors regarding stretching, imaging, and dye concentration, which is mediated by stretch. Consensus signals overcome such errors.
\item Describe the method for creating consensus signals using {\emph{M. florum}} data. Output ``maps'' are termed ``FCscans''. Validation of consensus method—perhaps tables showing statistical measures of success.
\item Are FCscans reproducible? Provide descriptions of tests that speak to reproducibility - use {\emph{M. florum}} data and analysis here.
\item How unique are FCscans? Present tests, in {\emph{M. florum}} that establish a quantitative measure of uniqueness. Here we don’t want to know the underlying sequence. We simply want a test of FCscan vs. FCscan. We will later discern sequence underpinnings to this important question. Permutations tests, etc.
\end{itemize}
\item Scaling to the human genome. 
\begin{itemize}
\item Ribosomal work - for reproducibility of FCscans
\item How sequence affects signals.
\item Heterozygosity
\end{itemize}
\end{enumerate}

\section*{Discussion}
We’ll write this after the results are in. Here we contextualize our findings with the worlds of statistics, genomics, and physical chemistry.

\section*{Conslusion}

\section*{Methods}
\begin{itemize}
\item Chemistries, imaging, preprocessing
\item Computational techniques, statistical proofs, etc
\end{itemize}

\section*{Acknowledgements}


\newpage
\bibliography{bibTex_Reference}
%\bibliography{research}

\end{document}
