%\documentclass[10pt,dvipsnames,table, handout]{beamer} % To printout the slides without the animations
\documentclass[10pt,dvipsnames,table]{beamer} 
%\usetheme{Luebeck} 
\usetheme{Madrid} 
%\usetheme{Marburg} 
\setbeamercolor{structure}{fg=cyan!90!white}
\setbeamercolor{normal text}{fg=white, bg=black}

%%%%%%%%%%%%%%%%%%%%%%%% Packages %%%%%%%%%%%%%%%%%%%%%%%%
\usepackage{amscd}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{amsthm}
\usepackage{amsxtra}
\usepackage{bbold}
%\usepackage{bigints}
\usepackage{color}
\usepackage{dsfont}
\usepackage{enumerate}
\usepackage[mathscr]{eucal}
%\usepackage{fancyhdr}
\usepackage{float}
%\usepackage{fullpage} %% Dont use this for beamer presentations
\usepackage{geometry}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{indentfirst}
\usepackage{latexsym}
\usepackage{listings}
\usepackage{lscape}
\usepackage{mathtools}
\usepackage{microtype}
\usepackage{natbib}
\usepackage{pdfpages}
\usepackage{verbatim}
\usepackage{wrapfig}
\usepackage{xargs}
\usepackage{xcolor}
\DeclareGraphicsExtensions{.pdf,.png,.jpg, .jpeg}

%%%%%%%%%%%%%%%%%%%%%%%% Commands %%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\Sup}{\textsuperscript}
\newcommand{\Exp}{\mathds{E}}
\newcommand{\Prob}{\mathds{P}}
\newcommand{\Z}{\mathds{Z}}
\newcommand{\Ind}{\mathds{1}}
\newcommand{\A}{\mathcal{A}}
\newcommand{\F}{\mathcal{F}}
\newcommand{\G}{\mathcal{G}}
\newcommand{\I}{\mathcal{I}}
\newcommand{\be}{\begin{equation}}
\newcommand{\ee}{\end{equation}}
\newcommand{\bes}{\begin{equation*}}
\newcommand{\ees}{\end{equation*}}
\newcommand{\union}{\bigcup}
\newcommand{\intersect}{\bigcap}
\newcommand{\Ybar}{\overline{Y}}
\newcommand{\ybar}{\bar{y}}
\newcommand{\Xbar}{\overline{X}}
\newcommand{\xbar}{\bar{x}}
\newcommand{\betahat}{\hat{\beta}}
\newcommand{\Yhat}{\widehat{Y}}
\newcommand{\yhat}{\hat{y}}
\newcommand{\Xhat}{\widehat{X}}
\newcommand{\xhat}{\hat{x}}
\newcommand{\E}[1]{\operatorname{E}\left[ #1 \right]}
%\newcommand{\Var}[1]{\operatorname{Var}\left( #1 \right)}
\newcommand{\Var}{\operatorname{Var}}
\newcommand{\Cov}[2]{\operatorname{Cov}\left( #1,#2 \right)}
\newcommand{\N}[2][1=\mu, 2=\sigma^2]{\operatorname{N}\left( #1,#2 \right)}
\newcommand{\bp}[1]{\left( #1 \right)}
\newcommand{\bsb}[1]{\left[ #1 \right]}
\newcommand{\bcb}[1]{\left\{ #1 \right\}}
\newcommand*{\permcomb}[4][0mu]{{{}^{#3}\mkern#1#2_{#4}}}
\newcommand*{\perm}[1][-3mu]{\permcomb[#1]{P}}
\newcommand*{\comb}[1][-1mu]{\permcomb[#1]{C}}


%%%%%%%%%%%%% For explanatory bubbles, use the following code %%%%%%%%%%%%%
%% \usepackage{tikz} %% For explanatory bubbles
%% \usepackage{xparse}
%% \usetikzlibrary{shapes.callouts,ocgx}

%% \newcommand{\tikzmark}[1]{\tikz[overlay,remember picture,baseline=0.5ex] \node (#1) {};}

%% % \explainword: #1= identifier to mark the word, #2 text
%% \NewDocumentCommand{\explainword}{r[] m}{
%%     \switchocg{#1}{#2}\tikzmark{#1}
%% }

%% \tikzset{my callout style/.style={
%%         draw,rectangle callout,anchor=pointer,callout relative pointer={(230:1cm)},
%%         rounded corners,align=center,text width=2cm,fill=cyan!20, 
%%     }
%% }

%% % \mycallout: #1 opacity style, #2 pointer base position, #3= text
%% \NewDocumentCommand{\mycallout}{O{opacity=0.8,text opacity=1} m m}{%
%% \begin{tikzpicture}[remember picture, overlay]
%%  \begin{scope}[ocg={ref=#2,status=invisible,name={#3}}]
%% \node[my callout style,#1]at (#2) {#3};
%% \end{scope}
%% \end{tikzpicture}
%% }
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%% TITLE PAGE %%%%%%%%%%%%%%%%%%%%%%%%
\DeclarePairedDelimiter\ceil{\lceil}{\rceil}
\title[Clustering gene-expression data]{Comparing different clustering techniques in analyzing gene-expression data}
\author{Subhrangshu Nandi}
\institute[Stat 760]{Stat 760, Fall 2014 \\
  Department of Statistics \\
 University of Wisconsin-Madison}
\date{December 11, 2014}

\begin{document}
\setlength{\baselineskip}{16truept}
\frame{\maketitle}

%%%%%%%%%%%% Slide 1 %%%%%%%%%%%%
%\begin{frame}
%\frametitle{Outline}
%\begin{itemize}
%\pause \item Introduction to functional data
%\pause \item Motivation of potential problem
%\pause \item Objective: Minimize effets of smoothing in two sample tests
%\pause \item Problem definition and two sample test
%\pause \item Some theoretical results
%\pause \item Computational results
%\pause \item Summary
%\end{itemize}
%\end{frame}

%%%%%%%%%%%% Slide 1 %%%%%%%%%%%%
\begin{frame}
\frametitle{Objective}
\begin{enumerate}
\pause \item Learn different clustering techniques
\begin{itemize}
\item K-means
%\item K-mediods (or Partitioning around \explainword[mediods]{mediods} - PAM)
\item K-mediods (or Partitioning around mediods - PAM)
\item Agglomerative Hierarchical
\item Self-organizing tree algorithm (SOTA) 
\item Gaussian mixture 
\end{itemize}
\pause \item Learn different metrics to compare clusters
\begin{itemize}
\item Connectivity index
\item Dunn index
\item Silhouette width
\end{itemize}
\pause \item Apply to gene-expression data and evaluate different clusters using these indices
\end{enumerate}

%% \mycallout{mediods}{Always members of the dataset}
%% \tikzset{my callout style/.append style={fill=orange!20}}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 2 %%%%%%%%%%%%
\begin{frame}
\frametitle{Clustering Techniques}
\begin{enumerate}
\pause \item K-means
\pause \item PAM $|$ Partitioning around mediods $|$ K-mediods
\begin{itemize}
\item Mediod: Minimal average dissimilarity to all objects in the cluster
\item Medoids are always members of the data set
\item More appropriate in the gene-expression context
\end{itemize}
\pause \item Agglomerative Hierarchical
\begin{itemize}
\item Yields a dendrogram (Exam 3!!)
\item Can be cut at a chosen height to produced desired number of clusters
\end{itemize}
\pause \item SOTA $|$ Self Organizing Tree Algorithm
\begin{itemize}
\item Divisive (top-down) hierarchical binary tree approach
\item Popular with analyzing gene-expression data
\item Dopazo and Carazo (1997)
\end{itemize}
\pause \item Gaussian mixture
\end{enumerate}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 3 %%%%%%%%%%%%
\begin{frame}
\frametitle{Metrics for Comparing Clusters}
\begin{itemize}
\pause \item Connectivity
\begin{itemize}
\item Measures if close neighbors are in the same cluster
\item If $\mathcal{C} = \{C_1, \dots, C_K \}$ of $N$ observations, then $Con(\mathcal{C}) = \sum\limits_{i=1}^{N} \sum\limits_{j=1}^{L} d_{i, n_j}$
\item $Con(\mathcal{C}) \in (0, \infty)$, should be \textcolor{green}{minimized} 
\end{itemize}
\pause \item Dunn index
\begin{itemize}
\item Ratio of the smallest {\emph{between cluster}} distance to the largest {\emph{within cluster}} distance
\item $Dunn(\mathcal{C}) \in (0, \infty)$, should be \textcolor{green}{maximized} 
\end{itemize}
\pause \item Silhouette width
\begin{itemize}
\item Degree of confidence in the clustering assignment of a particular observation
\item For observation $i$, $S(i) = \frac{b_i - a_i}{\max(b_i,a_i)}$, where, \\
$a_i = \frac{1}{|C(i)|}\sum\limits_{j \in C(i)}d(i,j)$, {\footnotesize (avg dist from objects in same cluster)} \\
$b_i = \min \limits_{C_k \in \mathcal{C} \backslash C(i) }\sum\limits_{j \in C_k}\frac{d(i,j)}{|C_k|}$, {\footnotesize(avg dist from objects in neighboring clusters)} \\
\item $S_{avg} \in [-1, 1]$. Well clustered observations have values closet to \textcolor{green}{1}, poorly clustered close to \textcolor{red}{-1}
\end{itemize}
\pause \item BIC (for model based clustering) (Raftery, 1999; Fraley and Raftery, 2002)
\end{itemize}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 4 %%%%%%%%%%%%
\begin{frame}
\frametitle{Dataset}
\begin{itemize}
\item Gene expresion data\footnotemark from 60 bone marrow samples of patients with the following types of Leukemia (12 in each group)
\begin{enumerate}
\item Acute Lymphoblastic Leukemia (ALL)
\item Acute Myeloid Leukemia (AML)
\item Chronic Lymphocytic Leukemia (CLL)
\item Chronic Myeloid Leukemia (CML)
\item \textcolor{green}{Non-leukemia and (healthy bone marrow) (NoL)}
\end{enumerate}
\pause \item All samples were obtained from untreated patients at the time of diagnosis
\pause \item Microarray gene expression data - \textcolor{red}{20,172 genes} \\
\hspace{0.5cm} - Each subset has design matrix 12 rows X 20,172 columns
\end{itemize}
\pause 
\begin{block}{GOAL} 
Use different clustering techniques and check if the 5 groups cluster differently.
\end{block}
\footnotetext[1]{Bioinformatics and Functional Genomics Group, Cancer Research Center, Salamanca. Spain}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 5 %%%%%%%%%%%%
\begin{frame}
\frametitle{Computation Time Comparison}
\begin{center}
\includegraphics[scale=0.4]{TimePlot_NoL.pdf}
\end{center}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 10 %%%%%%%%%%%%
\begin{frame}
\frametitle{NoL}
\begin{columns}
\begin{column}{5cm}
\includegraphics[scale=0.35]{Plot_NoL.jpeg} \\
%\pause \includegraphics[scale=0.25]{Plot_NoL_Scaled.jpeg}
\end{column}
\begin{column}{5cm}
\pause \includegraphics[scale=0.28]{BICPlot_NoL.pdf} \\
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 10a %%%%%%%%%%%%
\begin{frame}
\frametitle{NoL (continued)}
\begin{table}[H]
\centering
\begin{tabular}{l|r|r}
\hline
Metric & Type of Cluster & Cluster Size \\ 
\hline
Connectivity & hierarchical & 6 \\
Dunn & hierarchical & 6 \\
Silhouette & hierarchical & 8 \\
\hline
\end{tabular}
\caption{Recommendations for NoL}
\end{table}

\begin{table}[H]
\centering
\begin{tabular}{l|r|r}
\hline
Metric & Type of Cluster & Cluster Size \\ 
\hline
Connectivity & hierarchical & 6 \\
Dunn & SOTA & 9 \\
Silhouette & hierarchical & 6 \\
\hline
\end{tabular}
\caption{Recommendations for NoL (Scaled)}
\end{table}

\begin{block}{GMM}
9 clusters, ``ellipsoidal, varying volume, shape, and orientation''
\end{block}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 6 %%%%%%%%%%%%
\begin{frame}
\frametitle{ALL}
\begin{columns}
\begin{column}{5cm}
\includegraphics[scale=0.35]{Plot_ALL.jpeg} \\
%\pause \includegraphics[scale=0.25]{Plot_NoL_Scaled.jpeg}
\end{column}
\begin{column}{5cm}
\pause \includegraphics[scale=0.28]{BICPlot_ALL.pdf} \\
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 7 %%%%%%%%%%%%
\begin{frame}
\frametitle{AML}
\begin{columns}
\begin{column}{5cm}
\includegraphics[scale=0.35]{Plot_AML.jpeg} \\
%\pause \includegraphics[scale=0.25]{Plot_NoL_Scaled.jpeg}
\end{column}
\begin{column}{5cm}
\pause \includegraphics[scale=0.28]{BICPlot_AML.pdf} \\
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 8 %%%%%%%%%%%%
\begin{frame}
\frametitle{CLL}
\begin{columns}
\begin{column}{5cm}
\includegraphics[scale=0.35]{Plot_CLL.jpeg} \\
%\pause \includegraphics[scale=0.25]{Plot_NoL_Scaled.jpeg}
\end{column}
\begin{column}{5cm}
\pause \includegraphics[scale=0.28]{BICPlot_CLL.pdf} \\
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 9 %%%%%%%%%%%%
\begin{frame}
\frametitle{CML}
\begin{columns}
\begin{column}{5cm}
\includegraphics[scale=0.35]{Plot_CML.jpeg} \\
%\pause \includegraphics[scale=0.25]{Plot_NoL_Scaled.jpeg}
\end{column}
\begin{column}{5cm}
\pause \includegraphics[scale=0.28]{BICPlot_CML.pdf} \\
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 11 %%%%%%%%%%%%
\begin{frame}
\frametitle{Comparison between different types of Leukemia}
\begin{columns}
\begin{column}{.5\textwidth}
% latex table generated in R 3.1.2 by xtable 1.7-4 package
% Wed Dec 10 18:00:07 2014
\begin{table}[ht]
\centering
{\footnotesize
\rowcolors{1}{}{gray}
\begin{tabular}{rlll}
\hline
\multicolumn{4}{c}{Original Data} \\
  \hline
 Score & Cluster & Index & Subset \\ 
  \hline
  1078.35 & hier, 6 & Conn & ALL \\ 
  0.02 & hier, 6 & Dunn & ALL \\ 
  0.50 & hier, 6 & Silhouette & ALL \\ 
  1363.35 & hier, 6 & Conn & AML \\ 
  0.03 & hier, 6 & Dunn & AML \\ 
  0.49 & hier, 6 & Silhouette & AML \\ 
  904.49 & hier, 6 & Conn & CLL \\ 
  0.02 & hier, 6 & Dunn & CLL \\ 
  0.54 & hier, 6 & Silhouette & CLL \\ 
  1145.28 & hier, 6 & Conn & CML \\ 
  0.03 & hier, 6 & Dunn & CML \\ 
  0.47 & hier, 6 & Silhouette & CML \\ 
  555.36 & hier, 6 & Conn & NoL \\ 
  0.03 & hier, 6 & Dunn & NoL \\ 
  0.48 & hier, 8 & Silhouette & NoL \\ 
   \hline
\end{tabular}
}
\end{table}
\end{column}
\begin{column}{.5\textwidth}
% latex table generated in R 3.1.2 by xtable 1.7-4 package
% Wed Dec 10 18:00:31 2014
\pause
\begin{table}[ht]
\centering
{\footnotesize
\rowcolors{2}{}{gray}
\begin{tabular}{rll}
\hline
\multicolumn{3}{c}{Centered and Scaled} \\
\hline
Score & Clusters & Index \\ 
  \hline
157.43 & hier, 6 & Conn \\ 
  0.03 & hier, 6 & Dunn \\ 
  0.47 & hier, 9 & Silhouette \\ 
  927.66 & hier, 6 & Conn \\ 
  0.04 & hier, 8 & Dunn \\ 
  0.41 & hier, 6 & Silhouette \\ 
  853.40 & hier, 6 & Conn \\ 
  0.02 & hier, 6 & Dunn \\ 
  0.53 & hier, 6 & Silhouette \\ 
  966.44 & hier, 6 & Conn \\ 
  0.03 & hier, 7 & Dunn \\ 
  0.49 & hier, 6 & Silhouette \\ 
  938.19 & hier, 6 & Conn \\ 
  0.02 & sota, 9 & Dunn \\ 
  0.49 & hier, 6 & Silhouette \\ 
   \hline
\end{tabular}
}
\end{table}
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 11 %%%%%%%%%%%%
\begin{frame}
\frametitle{Comparison between different types of Leukemia (GMM)}
\begin{center}
\includegraphics[scale=0.45]{BICPlot_Comparison.pdf} \\
\end{center}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 11 %%%%%%%%%%%%
\begin{frame}
\frametitle{Comparison between different types of Leukemia}
\begin{columns}
\begin{column}{.5\textwidth}
\includegraphics[scale=0.35]{ComparisonPlot.pdf} \\
\end{column}
\begin{column}{.5\textwidth}
\includegraphics[scale=0.35]{ComparisonPlot_Sc.pdf} \\
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 12 %%%%%%%%%%%%
\begin{frame}
\frametitle{Clustering patients by gene expression}
{\footnotesize
We know there are 5 groups of patients
\begin{table}[H]
\centering
\begin{tabular}{l|r|r}
\hline
Metric & Type of Cluster & Cluster Size \\ 
\hline
Connectivity & hierarchical & 4 \\
greenDunn & kmeans & 6 \\
Silhouette & hierarchical & 4 \\
\hline
\end{tabular}
\caption{Recommendations for number of patient clusters}
\end{table}
\pause 
GMM recommends 5 clusters, by BIC criterion, with the cluster sizes: 13, 12, 15, 10, 10
}
\begin{center}
\includegraphics[scale=0.25]{BICPlot_Patient.pdf} \\
\end{center}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% \begin{frame}
% \begin{theorem}
% (i) Let $m=1,\ldots,s_1.$ We consider two runs $C_i,C_j,i\neq j$ in the subdesign $C^{m}.$ Then for $k=1,\ldots,d$ and $x,y\in \textbf{Z}_{s_1^2}$ 
% $$
% Pr(C_{ik}=x, C_{jk}=y) =\begin{cases}
% \frac{1}{s_1^3(s_1-1)},& \ceil{\frac{x}{s_1}}\neq\ceil{\frac{y}{s_1}}\\
% 0,&o.w.
% \end{cases}
% $$
% (ii) The joint probability mass function for $c_{m_1,ik},c_{m_2,jk},m_1\neq m_2$ is 
% $$
% Pr(c_{m_1,ik}=x,c_{m_2,jk}=y) =\begin{cases}
% \frac{1}{s_1^4},& \ceil{\frac{x}{s_1}}\neq\ceil{\frac{y}{s_1}}\\
% \frac{1}{s_1^3(s_1-1)},&\ceil{\frac{x}{s_1}}=\ceil{\frac{y}{s_1}},x\neq y\\
% 0,&o.w.
% \end{cases}
% $$
% \end{theorem}
% \end{frame}

% \begin{frame}
% \begin{theorem}
% Consider two row $C_i,C_j,i\neq j$ in the slice $C^{(m)}.$ Then for $k,l=1,\ldots,d$ and $k\neq l$
% $$ Pr(c_{ik},c_{il},c_{jk},c_{jl}) =\begin{cases}
% \frac{1}{s_1^2s_2(s_2-1)(s_1-1)},& (i,j)\in H_1\\
% \frac{1}{s_1^2s_2^2(s_1-1)},&(i,j)\in H_2.
% \end{cases}
% $$
% \end{theorem}
% \begin{theorem}
% \begin{itemize}
% \item $Var(\hat{\mu}_m) = s_1^{-1}\sum_{|u|\geq3}Var\{f_{m,u}(X)\} + o(s_1^{-1})$
% \item $Var(\hat{\mu}) = s_1^{-2}\sum_{|u|\geq3}Var\{f_u(X)\} + o(s_1^{-2})$
% \end{itemize}
% \end{theorem}
% \end{frame}
% %%%%%%%%%%%%%


\end{document}

