%\documentclass[10pt,dvipsnames,table, handout]{beamer} % To printout the slides without the animations
\documentclass[10pt,dvipsnames,table]{beamer} 
%\usetheme{Luebeck} 
%\usetheme{Madrid} 
%\usetheme{Marburg} 
%\usetheme{Warsaw} 
%\setbeamercolor{structure}{fg=cyan!90!white}
%\setbeamercolor{normal text}{fg=white, bg=black}
\usetheme{CambridgeUS}
%\setbeamertemplate{enumerate items}[default]
\setbeamercolor{item projected}{fg=green,bg=red}
%\setbeamercolor{itemize subitem}{fg=blue}
%\setbeamercolor{structure}{fg=cyan!90!white}
%\setbeamercolor{normal text}{fg=white, bg=black}
\setbeamercolor{block title}{bg=red!80,fg=white}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Input header file 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\input{HeaderfileTexSlides}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% TITLE PAGE 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\DeclarePairedDelimiter\ceil{\lceil}{\rceil}
\title[Fluoroscanning]{Fluoroscanning: {\emph{Next Generation Precision Genomics}}}
\author[Nandi, et al]{S. Nandi, M.A. Newton, P. Ravindran, S. Goldstein, K. Potamousis, M. Place, D.C. Schwartz}
\institute[LMCG]{	Laboratory of Molecular and Computational Genomics \\
 			University of Wisconsin - Madison }
\date{April 21, 2017}

\begin{document}
\setlength{\baselineskip}{16truept}
\frame{\maketitle}

%%%%%%%%%%%%%% Slide 1 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Motivation}
{\Large{
Next generation genomic sciences need to build systems to
\begin{enumerate}
\item Analyze genomes of every individual of the population
\item Analyze the complexities of cancer genomes
\item Develop targeted and precise treatment options for complex diseases
\end{enumerate}
in an {\underline{economical}} and {\underline{parsimonious}} manner.
}}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Motivation: mp3 representation of a Genome \footnote{\tiny{David C. Schwartz}}}
\vspace{-0.5cm}
\begin{columns}[t]
\begin{column}{0.5\textwidth}
\vspace{-0.5cm}
\begin{figure}[t]
\includegraphics[scale=0.24]{Images/wav.png} 
\end{figure}

\vspace{-0.5cm}
\begin{itemize}
{\footnotesize{
\item Wave is an uncompressed or ``lossless'' format
\item High sampling frequency of sound waves; high quality of music
\item Extremely large file size
}}
\end{itemize}
\vspace{-0.5cm}
\begin{figure}[H]
\includegraphics[scale=0.18]{Images/200114_Genome_650.jpg} 
\end{figure}

\end{column}

\begin{column}{0.5\textwidth}
\vspace{-0.5cm}
\begin{figure}[t]
\includegraphics[scale=0.18]{Images/mp3.jpg}
\end{figure}

\vspace{-0.5cm}
\begin{itemize}
{\footnotesize{
\item MP3 is a compressed or ``lossy'' format
\item Reduces the signal to its most necessary components
\item Very manageable file size; Easily transferable
}}
\end{itemize}
\vspace{-0.5cm}
\begin{figure}[H]
\hspace{-1cm}
\includegraphics[scale=0.3]{Plots/chr13_frag1000_consensusOnly.pdf} \\
\vspace{-0.5cm} \hspace{0.5cm}
\includegraphics[scale=0.28]{Plots/chr13_frag1025_consensusOnly.pdf} \\
\vspace{-0.5cm} \hspace{-2cm}
\includegraphics[scale=0.3]{Plots/chr13_frag1050_consensusOnly.pdf} \\
\vspace{-0.5cm} \hspace{1cm}
\includegraphics[scale=0.3]{Plots/chr13_frag1052_consensusOnly.pdf} \\
\end{figure}

\end{column}
\end{columns}
\vspace{-0.5cm}
{\tiny{\footnote{\tiny{https://nasiirblog.wordpress.com/2014/05/22/wav-vs-mp3/, http://archive.cosmosmagazine.com/}} }}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\LARGE
\begin{block}{Fluoroscanning}
A system to extract the most necessary and informative components of genomic sequences from fluorescence intensity profiles (Fscans) of imaged DNA molecules. 
\end{block}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\includegraphics[scale=0.45]{Images/FluoroscanningWorkflow_v3.pdf}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Fluoroscanning: Specific Aims}
\begin{block}{Aim 1}
Fluoroscanning signals (Fscans) of genomic intervals are unique and reproducible
\end{block}
\vspace{0.5in}
\begin{block}{Aim 2}
Predict Fscans from underlying sequence compositions of genomic intervals
\end{block}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Examples of Fscans}
\begin{figure}[t]
\includegraphics[scale=0.34, page=2]{Plots/chr13_frag7465.pdf}
%\hspace{0.25cm}
\includegraphics[scale=0.34, page=2]{Plots/chr13_frag7491.pdf}
\end{figure}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Types of noise}
\begin{columns}[t]
\begin{column}{0.6\textwidth}
\begin{figure}[H]
\begin{center}
%\hspace{2cm} \vspace{-1cm}
\includegraphics[scale = 0.25]{Images/Outlier_1.jpg}
\end{center}
\caption{Molecules too close to each other}
\end{figure}

%\item Molecules not straight
\begin{figure}[H]
\begin{center}
\includegraphics[scale = 0.25]{Images/Outlier_4.jpg}
\end{center}
\caption{Molecules not straight}
\end{figure}

\end{column}
\begin{column}{0.4\textwidth}
%\item Too much overlap, and curves molecules
\begin{figure}[H]
\begin{center}
\includegraphics[scale = 0.2]{Images/Outlier_2.jpg}
\end{center}
\caption{Too much overlap, and curved molecules}
\end{figure}

%\item Fragments overlapping DNA molecules
\begin{figure}[H]
\begin{center}
\includegraphics[scale = 0.2]{Images/Outlier_3.jpg}
\end{center}
\caption{Fragments overlapping DNA molecules}
\end{figure}
\end{column}

%\item 
%\end{itemize}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Need preprocessing}
\begin{figure}[t]
\includegraphics[scale=0.25, page=2]{Plots/chr13_frag7465.pdf}
\end{figure}
\vspace{-0.5cm}
\begin{itemize}
\item Image processing to detect noisy molecules
\item Normalize fluorescence intensities of each fragment by its median
\item Smooth the fluorescence intensities and evaluate them at equidistant points
\end{itemize}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Analyzing Fscans: Need Registration}
\begin{columns}
\begin{column}{0.5\textwidth}
\begin{figure}[H]
\begin{center}
\includegraphics[scale=0.3,page=2]{Plots/MF_Frag15.pdf}
\end{center}
\caption{NMaps aligned to {\emph{M. florum}} Interval 15}
\end{figure}
\end{column}
\begin{column}{0.5\textwidth}
\begin{itemize}
%\footnotesize
\item Molecules are of different lengths
\vspace{1cm}
\item Features are not aligned
\vspace{1cm}
\item Estimating consensus is not simple
\end{itemize}
\end{column}
\end{columns}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Statistical challenges}
\begin{block}{Statistical challenges}
{\Large{The primary challenge is to build statistical tools to enable Fluoroscanning data analysis.}} \\This involves
\begin{itemize}
\item {\bf{Preprocessing}} the data by understanding the different sources of noise, and appropriately reducing their effect or controlling for their presence
\vspace{0.5cm}
\item {\bf{Estimating}} a consensus fluorescence intensity profile for each genomic interval and establishing their sequence dependence
\vspace{0.5cm}
\item Setting up a {\bf{hypothesis testing}} framework to test Fscans from different genomic intervals for dissimilarity
\end{itemize}
\end{block}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{The registration problem}

Details skipped ...

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Consensus Fscan (cFscan) Estimation by ``Registration''}
\vspace{-0.5cm}
\begin{figure}
\includegraphics[scale=0.44,page=4]{Plots/SeqVsConsensus_SelectedFew_Chr19.pdf}
\end{figure}
\end{frame}

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Consensus Fscan (cFscan) Estimation}
\vspace{-0.5cm}
\begin{figure}
\includegraphics[scale=0.24,page=4]{Plots/SeqVsConsensus_SelectedFew_Chr19.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.24,page=6]{Plots/SeqVsConsensus_SelectedFew_Chr19.pdf} \\
\includegraphics[scale=0.24,page=16]{Plots/SeqVsConsensus_SelectedFew_Chr19.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.24,page=27]{Plots/SeqVsConsensus_SelectedFew_Chr19.pdf}
\end{figure}
\end{frame}

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\Large
\begin{block}{Aim 1}
Are these cFscans unique and reproducible?
\end{block}
\end{frame}

%%%%%%%%%%%%%% Slide 1 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Similarity between consensus signals}
Similarity index $\rho(f, g)$ between two consensus Fscans $f$ and $g$:
\[ \rho(f, g) = \frac{\int _{S_{fg}}f'(s)g'(s) ds}{\int _{S_{fg}}f'(s)^2 ds \int _{S_{fg}}g'(s)^2 ds} \]
where, $S_{fg} = S_f \cap S_g$, the intersection of Sobolev spaces formed by the functions $f, g$.\\
\[-1 \leq \rho(f, g) \leq 1\] \\
Higher values of $\rho$ implies more {\emph{similarity}} between two consensus

\end{frame}

%%%%%%%%%%%%%% Slide 2 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Similarity between sequences of two intervals: Similarity between GC Profiles}
\begin{figure}
\centering
\includegraphics[width=0.42\textwidth]{Plots/chr19_frag2481_pg8.png}
\hspace{0.5cm}
\includegraphics[width=0.42\textwidth]{Plots/chr19_frag3756_pg8.png}
\caption{$\rho_{gc} = 0.84, \rho_{fscan} = 0.70$}
\end{figure}
\end{frame}

%%%%%%%%%%%%%% Slide 3 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Dissimilar GC profiles and Fscans}
\begin{figure}
\centering
\includegraphics[width=0.42\textwidth]{Plots/chr19_frag2481_pg8.png}
\hspace{0.5cm}
\includegraphics[width=0.42\textwidth]{Plots/chr19_frag5423_pg8.png}
\caption{$\rho_{gc} = -0.56, \rho_{fscan} = -0.21$}
\end{figure}

\end{frame}

%%%%%%%%%%%%%% Slide 4 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Pairwise comparison for uniqueness}
\begin{itemize}
\item In chromosome 19, there are 457 sub-intervals that are 10.3 kb long, with at least 15 Nmaps depth
\item $ \binom {457} {2} = 104,196 $ pairs to compare. 
\item Mixed effects model to estimate relationship between consensus Fscans ($\rho_{fscan}$) and GC profile scores $\rho_{gc}$:
\[ 
\rho_{fs_{i,j}} = \alpha_i + \gamma_j + \beta \rho_{gc_{i,j}} + \epsilon_{i,j}\ \ \ i, j = 1, \dots, 457,\ \ i \ne j
\]
$\alpha_i \sim \mathcal{N}(0, \tau_1^2), \ \ \gamma_j \sim \mathcal{N}(0, \tau_2^2), \ \ \epsilon_{i,j} \sim \mathcal{N}(0, \sigma^2)$ \\
$ \alpha_i \indep \gamma_j \indep \epsilon_{i,j}$
\end{itemize}

\end{frame}

%%%%%%%%%%%%%% Slide 7 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Scatterplot with intervals with signals}
\begin{figure}
\centering
\includegraphics[width=0.5\textwidth, page=1]{Plots/ScatterPlots.pdf}
\end{figure}

\begin{block}{Conclusion}
Unique genomic intervals yield unique Fscans!!
\end{block}

\end{frame}

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\Large
\begin{block}{Aim 2}
Can we predict Fscans for any genomic interval, from sequence composition?
\end{block}
\end{frame}

%%%%%%%%%%%%%% Slide 3 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Predicting Fscan from sequence features}
\vspace{-0.5cm}
\begin{figure}
\centering
\includegraphics[width=0.45\textwidth, page=7]{Plots/chr19_Interval2481_registered.pdf}
\end{figure}
\vspace{-0.5cm}
\begin{enumerate}
\item Used counts of g, c, a, t, 2-mers, 3-mers, 4-mers and 5-mers to fit consensus Fscans (cFscans)
\item Used Gaussian kernel, to account for effect of 400bp on either side of a pixel in addition to the 200bp in the middle
\item Fit machine learning and statistical models (example random forest, gradient boosting, etc)
\end{enumerate}

\end{frame}

%%%%%%%%%%%%%% Slide 3 %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Predicting Fscan from sequence features}
\tiny
% latex table generated in R 3.3.1 by xtable 1.8-2 package
% Fri Jan 20 09:59:44 2017
\begin{table}[ht]
\centering
\begin{tabular}{rrrrrrrrrrrrr}
  \hline
 & cFscan & ttc & acagt & caggg & atca & ccagc & gct & gaaag & tccg & acttc & ctcct & tcgtg \\ 
  \hline
  1  & 1.0043 & 5 & 0 & 0 & 2 & 0 & 2 & 0 & 0 & 0 & 0 & 0 \\ 
  2  & 0.9958 & 2 & 0 & 0 & 2 & 0 & 1 & 0 & 0 & 0 & 0 & 0 \\ 
  3  & 0.9933 & 2 & 0 & 0 & 2 & 1 & 3 & 0 & 0 & 0 & 0 & 0 \\ 
  4  & 0.9950 & 5 & 0 & 0 & 2 & 0 & 0 & 0 & 1 & 1 & 0 & 0 \\ 
  5  & 0.9984 & 5 & 0 & 0 & 2 & 0 & 2 & 0 & 0 & 1 & 1 & 0 \\ 
  6  & 1.0011 & 3 & 0 & 0 & 1 & 1 & 2 & 0 & 0 & 1 & 0 & 0 \\ 
  7  & 1.0023 & 1 & 0 & 0 & 0 & 1 & 4 & 0 & 0 & 0 & 0 & 0 \\ 
  8  & 1.0020 & 6 & 0 & 0 & 1 & 0 & 2 & 0 & 0 & 0 & 0 & 0 \\ 
  9  & 1.0002 & 1 & 0 & 0 & 1 & 2 & 4 & 0 & 0 & 0 & 0 & 0 \\ 
  10 & 0.9989 & 0 & 0 & 0 & 0 & 2 & 3 & 0 & 0 & 0 & 0 & 0 \\ 
  11 & 1.0007 & 9 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 \\ 
  12 & 1.0079 & 6 & 0 & 0 & 1 & 0 & 3 & 1 & 0 & 0 & 0 & 0 \\ 
  13 & 1.0192 & 7 & 1 & 0 & 0 & 0 & 2 & 0 & 0 & 1 & 0 & 0 \\ 
  14 & 1.0310 & 3 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 \\ 
  15 & 1.0402 & 2 & 0 & 1 & 1 & 0 & 3 & 0 & 0 & 0 & 0 & 0 \\ 
  16 & 1.0439 & 4 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 \\ 
  17 & 1.0394 & 4 & 0 & 0 & 0 & 0 & 3 & 0 & 0 & 0 & 0 & 0 \\ 
  18 & 1.0245 & 3 & 0 & 0 & 1 & 0 & 5 & 0 & 0 & 0 & 2 & 0 \\ 
  19 & 1.0030 & 9 & 0 & 0 & 0 & 0 & 2 & 0 & 0 & 0 & 1 & 0 \\ 
  20 & 0.9833 & 8 & 1 & 0 & 1 & 1 & 2 & 0 & 0 & 0 & 0 & 0 \\ 
  21 & 0.9732 & 6 & 0 & 0 & 1 & 0 & 2 & 1 & 0 & 0 & 0 & 0 \\ 
  22 & 0.9743 & 0 & 0 & 1 & 0 & 3 & 4 & 0 & 0 & 0 & 0 & 1 \\ 
  23 & 0.9829 & 7 & 0 & 0 & 0 & 0 & 2 & 0 & 0 & 2 & 1 & 0 \\ 
  24 & 0.9949 & 5 & 0 & 0 & 3 & 0 & 2 & 0 & 0 & 1 & 0 & 0 \\ 
  25 & 1.0079 & 3 & 0 & 0 & 1 & 0 & 3 & 0 & 0 & 0 & 0 & 0 \\ 
   \hline
\end{tabular}
\end{table}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
\frametitle{Predicting Fscan from Sequence - Set up}
\begin{itemize}
\item Total sample size: 21,972 intervals from 22 chromosomes
\item Counts of 1 to 5-mers
\item Training set: 90\%, Testing set: 10\%
\item $\rho_{c}:$ {\emph{Similarity}} between average training {\underline{c}}onsensus Fscan and testing sub-interval, $|\rho_{c}| \leq 1$
\item $\rho_{p}:$ {\emph{Similarity}} between average {\underline{p}}redicted Fscan and testing sub-interval, $|\rho_{p}| \leq 1$
\item $\gamma = \frac{\rho_{p} - \rho_{c}}{1 - \rho_{c}} $ {\emph{Similarity}} improvement or prediction improvement
\end{itemize}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
\frametitle{Predicting Fscan from Sequence - Results}
\vspace{-0.5cm}
\begin{figure}
\includegraphics[scale=0.22, page=3]{Plots/chr17_8028_0_predictedBySeed.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.22, page=3]{Plots//chr17_3253_0_predictedBySeed.pdf} \\
\includegraphics[scale=0.22, page=3]{Plots/chr19_5558_1_predictedBySeed.pdf}
\hspace{0.5cm}
\includegraphics[scale=0.22, page=3]{Plots/chr20_957_1_predictedBySeed.pdf}
\end{figure}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%% SLIDE 7 %%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
\frametitle{Prediction Improvement - Random Forest}
\vspace{-0.5cm}
\begin{figure}
\includegraphics[page=1, scale=0.4]{Plots/PredictionComparisonPlots_RFRun03.pdf}
\end{figure}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%% SLIDE 9 %%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
\frametitle{When prediction is worse than mean model}
Example: chr19, interval 961 \\
 $\rho_{p, RF} = -0.405$ \hspace{1.5in} $\rho_{p, GBM} = -0.516$
\begin{figure}
\includegraphics[page=3, scale=0.3]{Plots/chr19_961_0_predictedBySeed_RFRun03.pdf}
\includegraphics[page=3, scale=0.3]{Plots/chr19_961_0_predictedBySeed_GBMRun01.pdf}
\end{figure}
\end{frame}

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Discretizing cFscans}
\footnotesize
\vspace{-0.5cm}
\begin{figure}
\centering
\includegraphics[width=0.5\textwidth, page=7]{Plots/chr19_Interval2481_registered.pdf}
\end{figure}
\vspace{-0.5cm}
Consider a cFscan $f(x)$, where $x$ denotes pixels.
\begin{itemize}
\item At each pixel $x$, a discretized representation of $f(x)$ is $f(x) > 0, f'(x) > 0, f''(x) > 0$
\item The discretized representation consisted of eight possible states $A: (1,1,1), B:(1,0,1), C:(1,1,0), D:(1,0,0), E:(0,1,1), F:(0,0,1), G:(0,1,0), H:(0,0,0)$
\item 144-letter ``Sequence'': GHHFFFEEGGGGGEEEACCCCDDDBBFFEEGHFEEACCCCCCCDDDDBFF ...EECCCDDDBFEEACCDDBBFHHHHHFFFFEEEEGGCCDDBBHHFFEEEA
\end{itemize}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%% Slide x %%%%%%%%%%%%%%
\begin{frame}
\frametitle{Uniqueness of discretized cFscans}
\begin{itemize}
\item Randomly select 1,000 intervals from 21,972 intervals in the dataset
\item Randomly choose 4kb (20 states), 5kb (25 states), ..., 8kb (40 states) and count the empirical frequency in the rest of the interval
\end{itemize}

\begin{table}[H]
\centering
\caption{Uniqueness of cFscans segments} 
\begin{tabular}{c|c|c}
\hline
\hline
Length (states) & Length (kb) 	& Empirical frequency 	\\
\hline
20 		& 4.120		& 5.077 		\\
25		& 5.150		& 0.194			\\
30		& 6.180		& 0.005			\\
35		& 7.210		& 0.000			\\
40		& 8.240		& 0.000			\\
\hline
\hline
\end{tabular}
\end{table}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%% SLIDE 9 %%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
\frametitle{Ongoing work}
\begin{itemize}
\item Writing up manuscript
\item Compare predictions with regions already identified with SVs like ``deletions''
\item Using markov property, derive theoretical bounds for uniqueness lengths
\item Derive error bound for Fscan uniqueness lengths
\end{itemize}
\end{frame}

\end{document}

