%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% This is for Stat 992, Fall 2015 presentation on 11/20/2015
%% Parts of this talk are from Documentation/Schwartz_Grp_Mtg/forMeeting_2015-09-28/slides_2015-09-28.tex
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\documentclass[10pt,dvipsnames,table, handout]{beamer} % To printout the slides without the animations
\documentclass[10pt,dvipsnames,table, notes]{beamer} 

%\documentclass[notes]{beamer}       % print frame + notes
%\documentclass[notes=only]{beamer}   % only notes
%\documentclass{beamer}              % only frames

%\usetheme{Luebeck} 
\usetheme{Madrid} 
%\usetheme{Marburg} 
%\usetheme{Warsaw} 
%\setbeamercolor{structure}{fg=cyan!90!white}
\setbeamercolor{normal text}{fg=white, bg=black}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Input header file 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\input{HeaderfileTexSlides}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% TITLE PAGE 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\DeclarePairedDelimiter\ceil{\lceil}{\rceil}
\title[Curve Registration in Fluoroscanning]{Applications of Curve Registration to Fluoroscanning\\ {\emph{essential tool for precision genomics}}}
\author[S. Nandi]{Subhrangshu Nandi \\
%\institute[Stat 741]{Stat 741, Spring 2015 \\
Statistics PhD Student \\
\vspace{0.5cm}
\small{in collaboration with:} \\
\textcolor{yellow}{Laboratory of Molecular and Computational Genomics}, \\
University of Wisconsin-Madison \\
Advisors: Prof. Michael Newton, Prof. David Schwartz}
\date{November 20, 2015}

\begin{document}
\setlength{\baselineskip}{16truept}
\frame{\maketitle}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% This is for 2015-09-28 Schwartz group meeting:
%% 1. Introduction to fluoroscanning
%% 2. Some statistical challenges of fluoroscanning
%% 3. Quality score & improvement of consensus
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Outline for this presentation:
%% Motivation
%% Background - Sequencing
%% Background - Nanocoding
%% 


%%%%%%%%%%%% Slide 1 %%%%%%%%%%%%
\begin{frame}
\frametitle{Motivation - road to precision genomics}
\begin{figure}[T]
\includegraphics[scale=0.38]{Image_Motivation.pdf}
\end{figure}

\note{Problem was motivated by a cancer genome (Multiple Myeloma); \\ 
LMCG biophysics, chemistry; \\ Data analysis; \\ with the goal to complement current sequencing efforts, \\ and enhance precision genomics}

%Images from the following websites:
%www.medimoon.com
%www.medcitynews.com
%jamia.oxfordjournals.org
%www.geuvadis.org
%www.sciencenutshell.com
%www.imaging-git.com
\end{frame}

%%%%%%%%%%%% Slide 2 %%%%%%%%%%%%
\begin{frame}
\frametitle{Background: Genome sequencing}
Genome sequencing \footnote{http://www.genomenewsnetwork.org/} is figuring out the order of DNA nucleotides, or bases that make up an organism's DNA. The human genome is made up of over 3 billion of these nucleic acids. A DNA sequence that has been translated from life's {\emph{chemical}} alphabet into our alphabet of written letters might look like this:
\begin{figure}[H]
\includegraphics[scale=0.6]{Image_Sequence_1.jpg}
\end{figure}
\pause
\begin{figure}[H]
\includegraphics[scale=0.4]{Image_DNA.jpg}
\end{figure}

\note{}
\end{frame}

%%%%%%%%%%%% Slide 3 %%%%%%%%%%%%
\begin{frame}
\frametitle{Background: Nanocoding}
\begin{figure}[T]
\includegraphics[scale=0.45]{Image_Nanocoding.jpg}
\end{figure}

\note{
I will try to explain the work that spanned over 10 years, in one slide, so please bear with me.\\
(1) A restriction enzyme and dye are added to the DNA molecules \\
(2) Wherever the enzyme encounters the sequence GCTGAGG, a red dot is added, called punctates \\
(3) The labeled molecules are laid on a positively charged surface \\
(4) Images of the DNA backbone are captured \\
(5) Images are analyzed using proprietary computer vision softwares, and distances between punctates are estimated \\
(6) This is called a nanocode, similar to barcode
}
\end{frame}

%%%%%%%%%%%% Slide 4 %%%%%%%%%%%%
\begin{frame}
\frametitle{Background: Nanocoding}
\begin{figure}[T]
\includegraphics[scale=0.28]{molecule37_frame65.png}
\end{figure}
\end{frame}

%%%%%%%%%%%% Slide 5 %%%%%%%%%%%%
\begin{frame}
\frametitle{Nanocoding - brief background}
\begin{figure}[T]
\includegraphics[scale=0.85]{NMaps_Aligned.png}
\includegraphics[scale=0.25]{BarcodeImage.jpg}
\end{figure}

{\emph{M.florum}} reference map: \\
{\footnotesize{81.6 18.7 59.4 13.9 9.0 5.0 12.3 10.2 15.0 25.4 3.9 20.9 15.6 10.2 9.5 11.1 4.5 
13.7 26.3 38.3 2.1 31.0 19.1 3.6 32.2 41.3 9.8}} \textcolor{SpringGreen}{16.4 15.7 6.3 36.5 18.3} {\footnotesize{32.1 21.0 
3.3 14.3 51.3 16.0 17.9}}

Molecule $2064486\_0\_11$: NtBspQI N  {\bf{31.24 \textcolor{SpringGreen}{16.80  15.77  6.25  35.08  18.12}}}

\note{
The sequence of lengths between punctates is called a nanocode, akin to barcode\\
It is unique to molecule, and it can be used to determine which location on the genome the molecule is from\\
Example, below is a reference map from a bacteria genome, and a nanocoded molecule
}

\end{frame}

%%%%%%%%%%%% Slide 6 %%%%%%%%%%%%
\begin{frame}
\frametitle{Fluoroscanning: Intensity profiles of DNA molecules}
\begin{center}
\begin{tikzpicture}
  \node (img1) {\includegraphics[scale=0.28]{molecule251_frame68.png}};
  \pause
  \node (img2) at (img1.north)[yshift=-3cm] {\includegraphics[scale=0.25, page=1]{Frag38_1.pdf}};
\end{tikzpicture}
\end{center}

\note{
If we plot the pixel intensity of the molecules, between punctates, we observe these fluoresence intensity profiles. \\
The brighter the pixel, the higher the value of the profile.
}
\end{frame}

%%%%%%%%%%%% Slide 7 %%%%%%%%%%%%
\begin{frame}
\frametitle{Fluoroscanning}
\begin{columns}
\column{.5\textwidth} 
Dimer of oxazole yellow (YOYO-1) is the fluorescent dye added to the DNA molecules (intercalates) \\
\vspace{0.5cm}
\begin{figure}
\includegraphics[scale=0.3]{DyeMolecule.jpg}
\end{figure}

\pause
\column{.5\textwidth} 
Fluorescent properties of bound YOYO depends strongly on base sequence \\
\vspace{0.5cm}
{\emph{Quantum yield of YO bound to “GC” sequences is twice as large as those bound to “AT” sequences; binding constants also vary with sequence}}
\begin{figure}
\includegraphics[scale=0.55]{DNAMolecule.jpg}
\end{figure}
\end{columns}
\vspace{0.5cm}
http://www.rcsb.org/pdb/explore/jmol.do?structureId=108D
\end{frame}

%%%%%%%%%%%% Slide 8 %%%%%%%%%%%%
\begin{frame}
\frametitle{Fluoroscanning: Question 1}
{\bf{\Large{What are the signature fluoresence intensity profiles of different regions on the genome?}}}
\vspace{1cm}
\begin{enumerate}
\item[\textcolor{Apricot}{(A)}] \large{\textcolor{Apricot}{Align a DNA molecule with the reference genome}} 
\vspace{1cm}
\item[\textcolor{Apricot}{(B)}] \large{\textcolor{Apricot}{Detect large scale structural variation }}
\end{enumerate}
\end{frame}

%%%%%%%%%%%% Slide 9 %%%%%%%%%%%%
\begin{frame}
\frametitle{Alignment of fluorescence intensity profiles}
\begin{figure}
\includegraphics[scale=0.45, page=2]{Frag38_1.pdf}
\end{figure}
\note{
The fluoresence intensity profiles are inherently smooth functions because neighboring pixels contribute to the grey levels of the images. 
}

\end{frame}

%%%%%%%%%%%% Slide 10 %%%%%%%%%%%%
\begin{frame}
\frametitle{Distance metric}
Distance between two curves $f_i$ and $f_j$:
\[ \rho(f_i, f_j) = \frac{\int _{S_{ij}}f'_i(s)f'_j(s) ds}{\int _{S_{ij}}f'_i(s)^2 ds \int _{S_{ij}}f'_j(s)^2 ds} \]
where, $S_{ij} = S_i \cap S_j$, the intersection of Sobolev spaces formed by the functions $f_i, f_j$.\\
Properties of $\rho(f_i, f_j)$:
\begin{itemize}
\item $|\rho(f_i, f_j)| \leq 1$
\item $|\rho(f_i, f_j)| = 1 \Leftrightarrow \exists \ \ A \in \Real ^+, B \in \Real, \ni f_i = Af_j + B$
\item Absolute value preserved under affine transformations of functions \\
$\rho(f_i, f_j) = \text{sign}(A_i\cdot A_j) \rho(A_if_i + B_i, A_jf_j + B_j)$
\item Also preserved under affine transformations of the abscissa (x-axis or time axis)
\end{itemize}

\note{
In mathematics, a Sobolev space is a vector space of functions equipped with a norm that is a combination of $L^p-\text{norms}$ of the function itself and its derivatives up to a given order.
}

\end{frame}

%%%%%%%%%%%% Slide 11 %%%%%%%%%%%%
\begin{frame}
\frametitle{Registration example}
\begin{center}
%\begin{figure}
\colorbox{white}{\includegraphics[scale=0.55, page=20]{Warpings_Seed10_Lambda_1e-01.pdf}}
%\end{figure}
\end{center}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 12 %%%%%%%%%%%%
\begin{frame}
\frametitle{Registration details}
\begin{enumerate}
\item Let $x(t)$ be the true curve; $y(t)$ be the curve to register. 
\item Let $h(t)$ be the warping function; $y[h(t)]$ be the registered curve.
\item $y[h(t)]$ and and $x(t)$ differ only in terms of amplitude variation, i.e., their values are proportional to one another across the range of $t$
\item Then, PCA of the following order two matrix should reveal essentially one component (smallest eigen value $\approx 0$) \\
\[C(h) = 
\begin{bmatrix}
\int \{x(t) \}^2dt & \int x(t) y[h(t)]dt \vspace{0.5cm} \\ 
\int x(t) y[h(t)]dt & \int \{y[h(t)]\}^2dt
\end{bmatrix}
\]
\item Choose $h(t)$ that minimizes the eigen value
\end{enumerate}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%% Slide 13 %%%%%%%%%%%%
\begin{frame}
\frametitle{Estimation of the warping function}
\begin{enumerate}
\item $h(t)$ is a strictly {\emph{monotone increasing}} function; Hence {\emph{invertible}}.
\item Let $w(t) = \frac{h''(t)}{h'(t)}$ be the relative acceleration of the warping function. \\
This will be used to {\emph{constrain}} the warping
\item The objective function is 
\[ \text{MINEIG}_{\lambda}(h) = \text{MINEIG}(h) + \lambda \int \{w^{(m)}(t)\}^2 dt \]
which is similar to:
\[ F_{\lambda}(y,x|h) = \int \|y'(t) - x'\{h(t)\} \|^2 dt + \lambda \int \{w^{(m)}(t)\}^2 dt \]
Penalizing $w(t)$ ensures smoothness and monotonicity
\item Express $w(t)$ in terms of B-Spline expansion: 
\[w(t) = \sum \limits_{k=0}^{K} c_k B_k(t)\]
Solve for the coefficients $c_k$
\end{enumerate}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%% Slide 12 %%%%%%%%%%%%
\begin{frame}
\frametitle{Iterated registration}

\end{frame}

%%%%%%%%%%%% Slide 9 %%%%%%%%%%%%
\begin{frame}
\frametitle{Permutation test}

\end{frame}

%%%%%%%%%%%% Slide 9 %%%%%%%%%%%%
\begin{frame}
\frametitle{Simulation results}

\end{frame}

%%%%%%%%%%%% Slide 9 %%%%%%%%%%%%
\begin{frame}
\frametitle{M.Florum examples}

\end{frame}

%%%%%%%%%%%% Slide 9 %%%%%%%%%%%%
\begin{frame}
\frametitle{Work in progress}

\end{frame}

%%%%%%%%%%%% Slide 17 %%%%%%%%%%%%
\begin{frame}
\frametitle{Acknowledgements}
Images in my slides were borrowed from: \\
www.medimoon.com \\
www.medcitynews.com \\
jamia.oxfordjournals.org \\
www.geuvadis.org \\
www.sciencenutshell.com \\
www.imaging-git.com \\

\end{frame}

%%%%%%%%%%%% Slide 17 %%%%%%%%%%%%
\begin{frame}
\frametitle{Questions? Comments?}
Questions?
\end{frame}

\end{document}

